Framework/Kernel/src/LiveListenerInfo.cpp

```diff
@@ -0,0 +1,87 @@
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidKernel/LiveListenerInfo.h"
+
+#include <iosfwd>
+
+#include "MantidKernel/FacilityInfo.h"
+#include "MantidKernel/InstrumentInfo.h"
+#include "MantidKernel/Logger.h"
+
+#include <Poco/DOM/Element.h>
+
+namespace Mantid {
+namespace Kernel {
+namespace {
+// static logger object
+Logger g_log("InstrumentInfo");
+}
+
+/**
+ * Construct from Facility Info XML.
+ *
+ * @param elem :: The Poco::XML::Element to read the data from
+ */
+LiveListenerInfo::LiveListenerInfo(const Poco::XML::Element *elem) {
+  m_name = elem->getAttribute("name");
+  if (m_name.empty()) {
+    // throw std::runtime_error("Listener name is not defined");
+    g_log.error() << "Listener connection name is not defined. "
+                  << "This listener will not be selectable.\n";
+  }
+
+  m_address = elem->getAttribute("address");
+  if (m_address.empty()) {
+    // throw std::runtime_error("Listener address is not defined");
+    g_log.error() << "Listener address for " << m_name << " is not defined\n";
+  }
+
+  m_listener = elem->getAttribute("listener");
+  if (m_listener.empty()) {
+    // m_listener = inst->facility().liveListener();
+    g_log.error() << "Listener class for " << m_name << " is not defined\n";
+  }
+}
+
+/**
+ * Construct manually.
+ *
+ * @param listener Class name of specific listener to use
+ * @param address Address which listener should use to connect
+ * @param name Name designator for this listener connection info
+ */
+LiveListenerInfo::LiveListenerInfo(const std::string &listener,
+                                   const std::string &address,
+                                   const std::string &name)
+    : m_name(name), m_address(address), m_listener(listener) {}
+
+bool LiveListenerInfo::operator==(const LiveListenerInfo &rhs) const {
+  return (this->address() == rhs.address() &&
```

Comment:
I agree for the address. This should remain case sensitive. 

I do not think that the `listener` class name should be allowed to be case sensitive. But we're getting tight on time, and this is probably a good candidate for another issue anyway.